projects/aca-playwright-shared/src/page-objects/components/search/search-sorting-picker.components.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Page } from '@playwright/test';
+import { BaseComponent } from '.././base.component';
+
+export type SortByType = 'Relevance' | 'Title' | 'Filename' | 'Modified date' | 'Modifier' | 'Created date' | 'Size' | 'Type';
+export class SearchSortingPicker extends BaseComponent {
+  private static rootElement = '#aca-button-action-menu';
+
+  public actionMenu = this.page.locator('[data-automation-id="auto_header_content_id_$thumbnail"]');
+  public sortOrderButton = this.page.locator('#aca-button-sorting-menu');
+  public sortByDropdownExpanded = this.page.locator('.mat-menu-panel').first();
+  public sortByList = this.page.locator('.mat-menu-panel button');
+
+  constructor(page: Page, rootElement = SearchSortingPicker.rootElement) {
+    super(page, rootElement);
+  }
+
+  async waitForSortByDropdownToExpand(): Promise<void> {
+    await this.sortByDropdownExpanded.waitFor();
+  }
+
+  async isSortOrderButtonDisplayed(): Promise<boolean> {
+    return this.sortOrderButton.isVisible();
+  }
+
+  async isSortByDropdownExpanded(): Promise<boolean> {
+    return this.sortByDropdownExpanded.isVisible();
+  }
+
+  async clickSortByDropdown(): Promise<void> {
+    await this.sortOrderButton.click();
+    await this.waitForSortByDropdownToExpand();
+  }
+
+  async getSortByOptionsList(): Promise<string[]> {
+    let sortOptionsCount = await this.sortByList.count();
+    let sortByOptions: string[] = [];
+    for (let i = 1; i < sortOptionsCount; i++) {
+        let textContent = (await this.sortByList.nth(i).textContent()).trim();
+        sortByOptions.push(textContent);
+    }
+    sortByOptions.sort();
+    return sortByOptions;
+}
+
+  async sortBy(option: SortByType, direction: string): Promise<void> {
```

Comment:
can direction have type too? People can make mistakes by typing for example: 'ascending' instead of 'asc' etc. So can you limit type to just 'asc' and 'desc' possible values?